### DISCUSSION TITLE
Max attempts in jedis pipelining

### DISCUSSION THREAD

**Discussion by rahul-narayana** [xetorthio/jedis#2995] [Q&A]
Hi Team,

In jedis cluster we have a constructor argument for maxAttempts which helps in retrying in case any slots are unreachable. Do we have any such support in cluster pipelining. If not what is the recommended way to handle such failures in cluster pipelining

**Comment by sazzad16** [comment]
> Do we have any such support in cluster pipelining.

No.

> what is the recommended way to handle such failures in cluster pipelining

Avoid pipelining, i.e. just use JedisCluster.

**Reply by rahul-narayana** [reply]
Will these be silent failures wrapped in response object

**Reply by sazzad16** [reply]
It's quite complicated. If the slot is unavailable since the ClusterConnectionProvider is created, it could be JedisMovedDataException wrapped in Response object. If the connection becomes unavailable mid-pipelining, `sync()` would throw JedisConnectionException. There are many cases. I hope you understand the caveats.
